src/analysis/convex/caratheodory.lean

```diff
@@ -0,0 +1,202 @@
+  {x : E} (m : x ∈ convex_hull (↑t : set E)) :
+  ∃ (y : (↑t : set E)), x ∈ convex_hull (↑(t.erase y) : set E) :=
+begin
+  rw finset.convex_hull_eq at m,
+  obtain ⟨f, fpos, fsum, rfl⟩ := m,
+  obtain ⟨g, gcombo, gsum, gpos⟩ := exists_relation_sum_zero_pos_coefficient_of_dim_succ_lt_card h,
+  clear h,
+  let s := t.filter (λ z : E, 0 < g z),
+  have : s.nonempty,
+  { obtain ⟨x, hx, hgx⟩ : ∃ x ∈ t, 0 < g x := gpos,
+    refine ⟨x, mem_filter.mpr ⟨hx, hgx⟩⟩, },
+  obtain ⟨i₀, mem, w⟩ := s.exists_min_image (λ z, f z / g z) this,
+  have hg : 0 < g i₀ := by { rw mem_filter at mem, exact mem.2 },
+  have hi₀ : i₀ ∈ t := filter_subset _ mem,
+  let k : E → ℝ := λ z, f z - (f i₀ / g i₀) * g z,
+  have hk : k i₀ = 0 := by field_simp [k, ne_of_gt hg],
+  have ksum : ∑ e in t.erase i₀, k e = 1,
+  { calc ∑ e in t.erase i₀, k e = ∑ e in t, k e :
+      by conv_rhs { rw [← insert_erase hi₀, sum_insert (not_mem_erase i₀ t), hk, zero_add], }
+    ... = ∑ e in t, (f e - f i₀ / g i₀ * g e) : rfl
+    ... = 1 : by rw [sum_sub_distrib, fsum, ← mul_sum, gsum, mul_zero, sub_zero] },
+  refine ⟨⟨i₀, hi₀⟩, _⟩,
+  { rw finset.convex_hull_eq,
+    refine ⟨k, _, ksum, _⟩,
+    { simp only [and_imp, sub_nonneg, mem_erase, ne.def, subtype.coe_mk],
+      intros e hei₀ het,
+      by_cases hes : e ∈ s,
+      { have hge : 0 < g e := by { rw mem_filter at hes, exact hes.2 },
+        rw ← le_div_iff hge,
+        exact w _ hes, },
+      { calc _ ≤ 0   : mul_nonpos_of_nonneg_of_nonpos _ _ -- prove two goals below
+           ... ≤ f e : fpos e het,
+        { apply div_nonneg_of_nonneg_of_pos (fpos i₀ (mem_of_subset (filter_subset t) mem)) hg },
+        { simpa [mem_filter, het] using hes }, } },
+    { simp only [subtype.coe_mk, center_mass_eq_of_sum_1 _ id ksum, id],
+      calc ∑ e in t.erase i₀, k e • e = ∑ e in t, k e • e :
+        by conv_rhs { rw [← insert_erase hi₀, sum_insert (not_mem_erase i₀ t), hk, zero_smul, zero_add], }
+      ... = ∑ e in t, (f e - f i₀ / g i₀ * g e) • e : rfl
+      ... = _ : _,
+      simp only [sub_smul, sum_sub_distrib, mul_smul, ← smul_sum, gcombo, smul_zero, sub_zero,
+        center_mass, fsum, inv_one', id.def, one_smul], }, },
+end
+
+/--
+The convex hull of a finset `t` with `findim ℝ E + 1 < t.card` is equal to
+the union of the convex hulls of the finsets `t.erase x` for `x ∈ t`.
+-/
+lemma step [decidable_eq E] (t : finset E) (h : findim ℝ E + 1 < t.card) :
+  convex_hull (↑t : set E) = ⋃ (x : (↑t : set E)), convex_hull ↑(t.erase x) :=
+begin
+  apply set.subset.antisymm,
+  { intros x m',
+    obtain ⟨y, m⟩ := mem_convex_hull_erase h m',
+    exact mem_Union.2 ⟨y, m⟩, },
+  { convert Union_subset _,
+    intro x,
+    apply convex_hull_mono, simp, }
+end
+
+/--
+The convex hull of a finset `t` with `findim ℝ E + 1 < t.card` is contained in
+the union of the convex hulls of the finsets `t' ⊆ t` with `t'.card ≤ findim ℝ E + 1`.
+-/
+lemma shrink' (t : finset E) (k : ℕ) (h : t.card = findim ℝ E + 1 + k) :
```

Comment:
This name is not optimal

Reply:
I guess they are fine, because they are in a dedicated namespace and very much tailored to the end goal of this file.